File: web_admin/src/types/commentRecheck.ts
export type CommentRecheckStatus =
  | 'not_checked'
  | 'queued'
  | 'checking'
  | 'exists'
  | 'missing'
  | 'failed'
  | 'login_required'
  | 'captcha_required'

export type CommentRecheckItem = {
  id: number
  automationResultId: number
  taskId: number
  taskDate: string
  doctorId: number
  doctorName: string
  keywordId?: number | null
  keyword: string
  deviceName: string
  publishAccount: string
  commentContent: string
  videoLink?: string
  status: CommentRecheckStatus
  checkedAt?: string
  failReason?: string
}

export type CommentRecheckQuery = {
  doctorId?: number | ''
  keywordId?: number | ''
  status?: CommentRecheckStatus | ''
  keyword?: string
  startDate?: string
  endDate?: string
  page: number
  pageSize: number
}

export type CommentRecheckListResponse = {
  items: CommentRecheckItem[]
  total: number
}

export type StartCommentRecheckPayload = {
  ids: number[]
}

export type StartCommentRecheckByDateRangePayload = {
  startDate: string
  endDate: string
}

export type StartCommentRecheckResponse = {
  submitted: number
  skipped: number
  loginRequired: boolean
}

export type CommentRecheckLoginStatus = {
  loggedIn: boolean
  sessionId?: string
  qrCodeUrl?: string
  message?: string
}

export type ConfirmCommentRecheckLoginPayload = {
  sessionId?: string
}

File: api/app/schemas/comment_recheck.py
from datetime import date, datetime
from typing import Literal

from pydantic import BaseModel, Field

CommentRecheckStatus = Literal[
    "not_checked",
    "queued",
    "checking",
    "exists",
    "missing",
    "failed",
    "login_required",
    "captcha_required",
]


class CommentRecheckItemRead(BaseModel):
    id: int
    automation_result_id: int = Field(serialization_alias="automationResultId")
    task_id: int = Field(serialization_alias="taskId")
    task_date: date = Field(serialization_alias="taskDate")
    doctor_id: int = Field(serialization_alias="doctorId")
    doctor_name: str = Field(serialization_alias="doctorName")
    keyword_id: int | None = Field(default=None, serialization_alias="keywordId")
    keyword: str = ""
    device_name: str = Field(serialization_alias="deviceName")
    publish_account: str = Field(serialization_alias="publishAccount")
    comment_content: str = Field(serialization_alias="commentContent")
    video_link: str | None = Field(default=None, serialization_alias="videoLink")
    status: CommentRecheckStatus
    checked_at: datetime | None = Field(default=None, serialization_alias="checkedAt")
    fail_reason: str | None = Field(default=None, serialization_alias="failReason")


class StartCommentRecheckPayload(BaseModel):
    ids: list[int] = Field(min_length=1)


class StartCommentRecheckResponse(BaseModel):
    submitted: int
    skipped: int = 0
    login_required: bool = Field(default=False, serialization_alias="loginRequired")


class CommentRecheckLoginStatusRead(BaseModel):
    logged_in: bool = Field(serialization_alias="loggedIn")
    session_id: str | None = Field(default=None, serialization_alias="sessionId")
    qr_code_url: str | None = Field(default=None, serialization_alias="qrCodeUrl")
    message: str | None = None


class ConfirmCommentRecheckLoginPayload(BaseModel):
    session_id: str | None = Field(
        default=None,
        validation_alias="sessionId",
        serialization_alias="sessionId",
    )
